lib/blocs/notifications_cubit/notifications_cubit.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:bloc/bloc.dart';
+import 'package:dalal_street_client/config/get_it.dart';
+import 'package:dalal_street_client/grpc/client.dart';
+import 'package:dalal_street_client/proto_build/actions/GetNotifications.pb.dart';
+import 'package:equatable/equatable.dart';
+
+import '../../config/log.dart';
+
+part 'notifications_state.dart';
+
+class NotificationsCubit extends Cubit<NotificationsCubitState> {
+  NotificationsCubit() : super(NotificationsCubitInitial());
+
+  Future<void> getNotifications() async {
+    try {
+      final GetNotificationsResponse notifResponse =
+          await actionClient.getNotifications(
+        GetNotificationsRequest(),
+        options: sessionOptions(getIt()),
+      );
+
+      emit(GetNotifSuccess(notifResponse));
+    } catch (e) {
+      emit(GetNotifFailure(e.toString()));
```

Comment:
```suggestion
      emit(GetNotifFailure(failedToReachServer));
```
Always show this message in catch block. It's better to avoid showing grpc exceptions directly to user

Reply:
Do this everywhere